rllib/algorithms/callbacks.py

```diff
@@ -108,26 +108,36 @@ def on_sub_environment_created(
         pass
 
     @OverrideToImplementCustomLogic
-    def before_sub_environment_reset(
+    def on_episode_created(
         self,
         *,
         worker: "RolloutWorker",
-        sub_environment: EnvType,
+        base_env: BaseEnv,
+        policies: Dict[PolicyID, Policy],
         env_index: int,
+        episode: Union[Episode, EpisodeV2],
         **kwargs,
     ) -> None:
-        """Callback run before a sub-environment is reset.
+        """Callback run when a new episode is created (but has not started yet!).
+
+        This method gets called after a new Episode(V2) instance is created to
+        start a new episode. This happens before the respective sub-environment's
+        (usually a gym.Env) `try_reset()` is called by RLlib.
 
-        This method gets called before every `try_reset()` is called by RLlib
-        on a sub-environment (usually a gym.Env). This includes the very first (initial)
-        reset performed on each sub-environment.
+        episode
```

Comment:
is this a typo? should remove?

Reply:
can I suggest we also update the doc string for on_episode_start() to point out the difference between it and on_episode_created()? basically, on_episode_start() gets called after base_env.try_reset() is done.
since users may be reading about on_episode_start() without noticing the details here.

Reply:
Yeah, sorry, a leftover. :) Removed.

Cleaned up both docstrings and added exact sequence of events to both.